### PR TITLE
Bump spotless and versionlock

### PR DESCRIPTION
**Pull request by daberkow**
## Before this PR
Issues arose running spotless checks, as shown in #2828 

## After this PR
==COMMIT_MSG==
Update spotless java to 6.25.0 from 6.6.0
==COMMIT_MSG==

## Possible downsides?
It was noted that updates to spotless "Newer spotless versions have issues resolving dependencies at configuration time", but the build seems to be working.

### CODE REVIEW

**Comment by daberkow** [comment]
This has been tested locally against a repo showing the issue, and bumping did correct the issue. I realized 6.25.0 is out, and will bump again.

**Comment by daberkow** [comment]
I am doing some checks, 6.25.0 is passing in circle but failing on my local system. 6.24.0 is passing in both. Trying to figure out why.

File: versions.props [hunk 1]
```diff
@@ -45,8 +45,7 @@ org.mockito:* = 5.12.0
 # Don't upgrade, we will remove this in a future release.
 # Users should depend on this plugin directly such that it will get updated by excavator.
 com.palantir.javaformat:gradle-palantir-java-format = 1.1.0
-# Newer spotless versions have issues resolving dependencies at configuration time
-com.diffplug.spotless:spotless-plugin-gradle = 6.6.0
```
Comment (pkoenig10) [inline versions.props:49]: The issues that prompted this pin in https://github.com/palantir/gradle-baseline/pull/2312 is still open:

https://github.com/diffplug/spotless/issues/1380

Do we have reason to believe this issue is fixed?

Comment (daberkow) [reply versions.props:49]: In testing this and coil mentioned in the issue. Coil still has it, this repo seems to currently not have it, and people seem to think its related to indirect dependencies being loaded. Gradle does correct the issue if it comes up, it just causes a perf hit. Without bumping spotless, repos using OpenJFx and some other libs can compile at all. If we do not update this lib, repos with that issue will be forced to disable baseline-spotless entirely.

Comment (daberkow) [reply versions.props:49]: The bug reports say that gradle has a catch for this and the worst case scenario is: if there are indirect dependencies and it has to work through them, hitting this regression, then it is a small perf hit, but doesnt fail builds. Where as without this we cant compile UI apps at all. :/

Comment (daberkow) [reply versions.props:49]: Is there a path to get spotless updated here?

**Comment by daberkow** [comment]
https://github.com/palantir/gradle-baseline/pull/2313
https://github.com/diffplug/spotless/issues/1380 
With gradle-consistent-versions this will cause a full build failure. 

Pointed out, for anyone who has issues, you can add the newer spotless version to your classpath to override what is given here, and work from there.

**Pull request closed without merging**